agent/consul/server_address_lookup.go

```diff
@@ -0,0 +1,56 @@
+package consul
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/hashicorp/consul/agent/metadata"
+	"github.com/hashicorp/raft"
+)
+
+// ServerLookup encapsulates looking up servers by id and address
+type ServerLookup struct {
```

Comment:
I'd probably change this file to `server_lookup.go` and change the object receiver name to `sl` for all these methods.